sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/LineCounterTest.java

```diff
@@ -20,73 +20,50 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
-import java.util.Arrays;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
 import org.sonar.api.batch.sensor.internal.SensorContextTester;
 import org.sonar.api.measures.CoreMetrics;
-import org.sonar.api.measures.FileLinesContext;
-import org.sonar.api.measures.FileLinesContextFactory;
 import org.sonarsource.analyzer.commons.xml.ParseException;
 import org.sonarsource.analyzer.commons.xml.XmlFile;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
-import static org.mockito.Mockito.when;
 
 class LineCounterTest {
 
-  private FileLinesContextFactory fileLinesContextFactory;
-  private FileLinesContext fileLinesContext;
-
-  @BeforeEach
-  void setUp() {
-    fileLinesContextFactory = mock(FileLinesContextFactory.class);
-    fileLinesContext = mock(FileLinesContext.class);
-    when(fileLinesContextFactory.createFor(any(InputFile.class))).thenReturn(fileLinesContext);
-  }
-
   @Test
   void test_simple_file() throws IOException {
-    verifyMetrics("simple.xml", 1,
-      1, 4, 5, 6, 7, 8, 9, 10, 12, 13, 14, 15, 16, 17, 18);
+    verifyMetrics("simple.xml", 1, 15, """
+        {"ncloc_data":{"1":1,"4":1,"5":1,"6":1,"7":1,"8":1,"9":1,"10":1,"12":1,"13":1,"14":1,"15":1,"16":1,"17":1,"18":1}}""");
   }
 
   @Test
   void test_complex_file() throws IOException {
-    verifyMetrics("complex.xml", 17,
-      1, 2, 3, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 22, 25, 26, 27, 34, 35, 36, 37, 38, 39);
+    verifyMetrics("complex.xml", 17, 23, """
+      {"ncloc_data":{"1":1,"2":1,"3":1,"11":1,"12":1,"13":1,"14":1,"15":1,"16":1,"17":1,"18":1,"19":1,"20":1,"22":1,"25":1,"26":1,"27":1,"34":1,"35":1,"36":1,"37":1,"38":1,"39":1}}""");
   }
 
   @Test
   void test_invalid_file() {
-    assertThrows(ParseException.class, () -> verifyMetrics("invalid.xml", -1));
+    assertThrows(ParseException.class, () -> verifyMetrics("invalid.xml", -1, 0, ""));
   }
 
   @Test // SONARXML-19
   void test_file_with_char_before_prolog() throws Exception {
-    verifyMetrics("char_before_prolog.xml", 1, 3, 6, 7, 8, 9, 10, 11, 12, 13);
+    verifyMetrics("char_before_prolog.xml", 1, 9, """
+      {"ncloc_data":{"3":1,"6":1,"7":1,"8":1,"9":1,"10":1,"11":1,"12":1,"13":1}}""");
   }
 
-  private void verifyMetrics(String filename, int commentLinesNumber, int... linesOfCode) throws IOException {
+  private void verifyMetrics(String filename, int commentLinesNumber, int ncloc, String linesMetrics) throws IOException {
```

Comment:
In the tests above, it looks like the only metric we check is `ncloc_data`. But we also pass ncloc as argument, which also always matches the number of entries under `ncloc_data` here. Should we split the test with one checking the metrics using the JSON and a second one checking the measures stored in the sensor storage (`ncloc` and `comment_lines`)?

Reply:
Ideally we should be able to pass the sensor storage to the `FilesLinesContextTester` so that everything is stored in the same place but splitting the test method in two might be enough for now

Reply:
For now, I only removed `int ncloc` and compute it from the `FilesLinesContextTester` storage. And because I had to split in different commits, sorry, the latest changes are already merged into the second commits.